api/jobs_test.go

```diff
@@ -62,12 +67,80 @@ func TestCreateJobHandlerWithValidID(t *testing.T) {
 	})
 }
 
+func TestGetJobHandlerWithValidID(t *testing.T) {
+
+	Convey("Given a Search Reindex Job API that returns specific jobs using their id as a key", t, func() {
+
+		jobStoreMock := &mock.JobStoreMock{
+			GetJobFunc: func(ctx context.Context, id string) (models.Job, error) {
+				switch id {
+				case testJobID2:
+					return models.NewJob(testJobID2), nil
+				default:
+					return models.Job{}, errors.New("the job store does not contain the job id entered")
+				}
+			},
+		}
+
+		api := Setup(ctx, mux.NewRouter(), jobStoreMock)
+
+		Convey("When a request is made to get a specific job that exists in the Job Store", func() {
+			req := httptest.NewRequest("GET", fmt.Sprintf("http://localhost:25700/jobs/%s", testJobID2), nil)
+			resp := httptest.NewRecorder()
+
+			api.Router.ServeHTTP(resp, req)
+
+			Convey("Then the relevant search reindex job is returned with status code 200", func() {
+				So(resp.Code, ShouldEqual, http.StatusOK)
+				payload, err := ioutil.ReadAll(resp.Body)
+				So(err, ShouldBeNil)
+				jobReturned := models.Job{}
+				err = json.Unmarshal(payload, &jobReturned)
+				expectedJob := models.NewJob(testJobID2)
+
+				Convey("And the returned job resource should contain expected values", func() {
+					So(jobReturned.ID, ShouldEqual, expectedJob.ID)
+					So(jobReturned.Links, ShouldResemble, expectedJob.Links)
+					So(jobReturned.NumberOfTasks, ShouldEqual, expectedJob.NumberOfTasks)
+					So(jobReturned.ReindexCompleted, ShouldEqual, expectedJob.ReindexCompleted)
+					So(jobReturned.ReindexFailed, ShouldEqual, expectedJob.ReindexFailed)
+					So(jobReturned.ReindexStarted, ShouldEqual, expectedJob.ReindexStarted)
+					So(jobReturned.SearchIndexName, ShouldEqual, expectedJob.SearchIndexName)
+					So(jobReturned.State, ShouldEqual, expectedJob.State)
+					So(jobReturned.TotalSearchDocuments, ShouldEqual, expectedJob.TotalSearchDocuments)
+					So(jobReturned.TotalInsertedSearchDocuments, ShouldEqual, expectedJob.TotalInsertedSearchDocuments)
+				})
+
+			})
+
+		})
+
+		Convey("When a request is made to get a specific job that does not exist in the Job Store", func() {
+			req := httptest.NewRequest("GET", fmt.Sprintf("http://localhost:25700/jobs/%s", testJobID1), nil)
+			resp := httptest.NewRecorder()
+
+			api.Router.ServeHTTP(resp, req)
+
+			Convey("Then an empty search reindex job is returned with status code 404", func() {
+				So(resp.Code, ShouldEqual, http.StatusNotFound)
+				errMsg := strings.TrimSpace(resp.Body.String())
+				So(errMsg, ShouldEqual, "Failed to find job in job store")
+				payload, err := ioutil.ReadAll(resp.Body)
```

Comment:
Those lines are just confirming that an empty job is being returned (as well as the error). I'm happy to remove those lines if they're not adding value but, if I do, then I would also like to change the step to say "Then an error is returned with status code 404" instead of "Then an empty search reindex job is returned with status code 404".  Would you like me to do that?

Reply:
Forget that - I've just read your next comment so I'll change it as suggested.